lib/app-utils.js

```diff
@@ -504,16 +600,24 @@ async function unzipApp(appPathOrZipStream, depth = 0) {
 }
 
 /**
- * @this {import('./driver').XCUITestDriver}
+ * The callback invoked by configureApp helper
+ * when it is necessary to download the remote application.
+ * We assume the remote file could be anythingm, but only
+ * .zip and .ipa formats are supported.
+ * A .zip archive can contain one or more
+ *
+ * @this {XCUITestDriver}
  * @param {import('@appium/types').DownloadAppOptions} opts
  * @returns {Promise<string>}
  */
-export async function onDownloadApp({stream}) {
-  return await unzipApp.bind(this)(stream);
+export async function onDownloadApp({stream, headers}) {
+  return this.isRealDevice()
+    ? await downloadIpa.bind(this)(stream, headers)
```

Comment:
Should here return string of promise instead of IpaInfo?

Reply:
good catch, thanks. I've changed the return type of downloadIpa helper
weird that typescript did not catch that